### PR TITLE
add deviatoric stress postprocessing

### PR DESCRIPTION
**Pull request by jvanhunen**
This is a small addition to postprocess/visualisation: 
the stress.cc (and stress.h) postprocessor was outputting all components of the (full) stress tensor. 
I created a slightly modified set of files (devstress.cc and devstress.h) that output the deviatoric stress: for each component, the full stress = viscosity*strainrate + pressure, and the deviatoric stress is simply devstress = viscosity*strainrate (i.e. without the pressure term). 

This is useful to (e.g.) visualize the stresses that drive shortening or extension in the lithosphere.

### CODE REVIEW

File: source/postprocess/visualization/devstress.cc [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(DevStress,
+                                                  "devstress",
```
Comment (tjhei) [inline source/postprocess/visualization/devstress.cc]: we typically spell out names, so please use "deviatoric stress"

Comment (jvanhunen) [reply source/postprocess/visualization/devstress.cc]: I changed the following files names: 
source/postprocess/visualization/devstress.cc --> source/postprocess/visualization/deviatoric_stress.cc and 
include/aspect/postprocess/visualization/devstress.h --> include/aspect/postprocess/visualization/deviatoric_stress.h

Furthermore: 
- Class name: DevStress to DeviatoricStress
- Postprocessing name in parameter file is now "deviatoric_stress"
- visualisation option names changed from "devstress_xx" to deviatoric_stress_xx", and similar for all other components of stress tensor.

File: include/aspect/postprocess/visualization/devstress.h [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/*
```
Comment (gassmoeller) [inline include/aspect/postprocess/visualization/devstress.h]: I think you included the files twice, once as `devstress.h` and once as `deviatoric_stress.h`, same for the .cc files.

Comment (jvanhunen) [reply include/aspect/postprocess/visualization/devstress.h]: you are right: shear stress does exactly the same. So no need anymore for this postprocessor

**Pull request closed without merging**